plugin/path_creds.go

```diff
@@ -162,6 +162,32 @@ func (b *backend) generateAndReturnCreds(ctx context.Context, engineConf *config
 		return nil, err
 	}
 
+	var currentPassword, lastPassword string
+	if val, ok := previousCred["current_password"].(string); ok {
```

Comment:
If `current_password` or `last_password` are not found in the map then we are setting them to empty string in the call to `framework.PutWAL`. Is that a concern?

Reply:
I think the only concern is when we try to rollback the password if `PutWAL` fails. I can add additional logic to handle this situation.

An empty current/last password would roll back the role to it's initial state (no password since it hasn't been accessed yet).

Reply:
Done!